akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/MessageAdapterSpec.scala

```diff
@@ -269,17 +293,10 @@ class MessageAdapterSpec
       probe.expectMessage(3)
       // exception was thrown for 3
       probe.expectMessage("stopped")
+      probe.expectTerminated(ref)
 
-      deadLetterProbe.receiveMessage().message match {
-        case AdaptMessage(Pong(greeting), _) => greeting shouldBe "hello"
-        case Pong(greeting)                  => greeting shouldBe "hello"
-        case other                           => fail(s"Unexpected dead letter: [$other]")
-      }
-      deadLetterProbe.receiveMessage().message match {
-        case AdaptMessage(Pong(greeting), _) => greeting shouldBe "hello"
-        case Pong(greeting)                  => greeting shouldBe "hello"
-        case other                           => fail(s"Unexpected dead letter: [$other]")
-      }
```

Comment:
It was this that failed. From the logs I think it's clear that it got a DeadLetter from a previous test. See the class names for Pong. `MessageAdapterSpec$Pong$4` and `MessageAdapterSpec$Pong$7`

```
2021-09-21 00:29:16,755 INFO  akka.actor.LocalActorRef akkaDeadLetter - Message [akka.actor.typed.scaladsl.MessageAdapterSpec$Pong$4] wrapped in [akka.actor.typed.internal.AdaptMessage] to Actor[akka://MessageAdapterSpec/user/$f#1191629466] was not delivered. [4] dead letters encountered. If this is not an expected behavior then Actor[akka://MessageAdapterSpec/user/$f#1191629466] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'. MDC: {akkaAddress=akka://MessageAdapterSpec, sourceThread=MessageAdapterSpec-akka.actor.default-dispatcher-7, akkaSource=akka://MessageAdapterSpec/user/$f, sourceActorSystem=MessageAdapterSpec, akkaMessageClass=akka.actor.typed.scaladsl.MessageAdapterSpec$Pong$4, akkaTimestamp=00:29:16.754UTC}
9124
2021-09-21 00:29:16,755 ERROR akka.actor.SupervisorStrategy  - boom MDC: {akkaAddress=akka://MessageAdapterSpec, akkaSource=akka://MessageAdapterSpec/user, sourceActorSystem=MessageAdapterSpec}
9125
akka.actor.testkit.typed.TestException: boom
9126
2021-09-21 00:29:16,755 INFO  akka.actor.LocalActorRef akkaDeadLetter - Message [akka.actor.typed.scaladsl.MessageAdapterSpec$Pong$7] wrapped in [akka.actor.typed.internal.AdaptMessage] to Actor[akka://MessageAdapterSpec/user/$h#-173418177] was not delivered. [5] dead letters encountered. If this is not an expected behavior then Actor[akka://MessageAdapterSpec/user/$h#-173418177] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'. MDC: {akkaMessageClass=akka.actor.typed.scaladsl.MessageAdapterSpec$Pong$7, akkaAddress=akka://MessageAdapterSpec, sourceThread=MessageAdapterSpec-akka.actor.default-dispatcher-6, akkaTimestamp=00:29:16.755UTC, akkaSource=akka://MessageAdapterSpec/user/$h, sourceActorSystem=MessageAdapterSpec}
9127
2021-09-21 00:29:16,755 INFO  akka.actor.LocalActorRef akkaDeadLetter - Message [akka.actor.typed.scaladsl.MessageAdapterSpec$Pong$7] wrapped in [akka.actor.typed.internal.AdaptMessage] to Actor[akka://MessageAdapterSpec/user/$h#-173418177] was not delivered. [6] dead letters encountered. If this is not an expected behavior then Actor[akka://MessageAdapterSpec/user/$h#-173418177] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'. MDC: {akkaAddress=akka://MessageAdapterSpec, sourceThread=MessageAdapterSpec-akka.actor.default-dispatcher-6, akkaSource=akka://MessageAdapterSpec/user/$h, sourceActorSystem=MessageAdapterSpec, akkaMessageClass=akka.actor.typed.scaladsl.MessageAdapterSpec$Pong$7, akkaTimestamp=00:29:16.755UTC}
```

I could reproduce that with a few sleeps at strategic places.